src/battle/action_cmd/tidal_wave.c

```diff
@@ -2,12 +2,93 @@
 
 #define NAMESPACE action_command_tidal_wave
 
-INCLUDE_ASM(s32, "battle/action_cmd/tidal_wave", func_802A9000_42C6F0);
+extern s32 D_80108AFC;
+extern s32 D_80108B28;
+extern s32 D_802944A0;
 
-INCLUDE_ASM(s32, "battle/action_cmd/tidal_wave", func_802A9138_42C828);
+ApiStatus N(CreateHudElements)(ScriptInstance* script, s32 isInitialCall) {
+    HudElement* hudElement;
+    s32 i;
+    ActionCommandStatus* actionCommandStatus = &gActionCommandStatus;
+    BattleStatus* battleStatus = &gBattleStatus;
+
+    battleStatus->unk_82 = 5;
+    battleStatus->unk_434 = &D_802944A0;
+    
+    if (battleStatus->unk_83 == 0) {
+        battleStatus->actionSuccess = 0;
+        return ApiStatus_DONE2;
+    } else {
+        func_80268858();
+        actionCommandStatus->actionCommandID = 0x17;
+        actionCommandStatus->state = 0;
+        actionCommandStatus->unk_60 = 0;
+        actionCommandStatus->unk_44 = 0;
+        actionCommandStatus->unk_48 = 0;
+        battleStatus->unk_84 = 0;
+        actionCommandStatus->hudElementX = -0x30;
+        actionCommandStatus->hudElementY = 0x50;
+        hudElement = create_hud_element(&D_80108AFC);
+        actionCommandStatus->hudElements[0] = hudElement;
+        set_hud_element_render_pos(hudElement, actionCommandStatus->hudElementX, actionCommandStatus->hudElementY + 0x1C);
+        set_hud_element_render_depth(hudElement, 0);
+        set_hud_element_flags(hudElement, 0x82);
+
+        for (i = 1; i < 15; i++) {
```

Comment:
```suggestion
        for (i = 1; i < ARRAY_COUNT(actionCommandStatus->hudElements); i++) {
```

Reply:
This might not be quite equivalent - There's enough space left in the struct for the hudElements array to have one more element than currently listed, which would break this if that's the case. I'd hold off on implementing this change until we are absolutely sure on the length of the hudElements array.